packages/peregrine/lib/talons/CheckoutPage/GuestSignIn/useGuestSignIn.js

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useState } from 'react';
+
+export const useGuestSignIn = props => {
+    const { toggleActiveContent } = props;
+    const [view, setView] = useState('SIGNIN');
+
+    const toggleForgotPasswordView = useCallback(() => {
+        setView(currentView =>
+            currentView === 'SIGNIN' ? 'FORGOT_PASSWORD' : 'SIGNIN'
+        );
+    }, []);
+
+    const toggleCreateAccountView = useCallback(() => {
+        setView(currentView =>
+            currentView === 'SIGNIN' ? 'CREATE_ACCOUNT' : 'SIGNIN'
+        );
+    }, []);
+
+    const handleBackToCheckout = useCallback(() => {
+        toggleActiveContent();
+        setView('SIGNIN');
```

Comment:
It was a little strange to see us setting the view here but this is just setting us up for next time we toggle the active content, right?

Reply:
Correct. If you ever leave the view it should reset back to the sign-in form.